### ISSUE TITLE
Minus operator spacing issue with function-calc-no-unspaced-operator

### ISSUE THREAD

**Issue by perrin4869** [stylelint/stylelint#1858]
In the `calc` function, there is an error when using the minus operator to negate a variable, i.e., `calc(-var(--foo))` instead of `calc(-1 * var(--foo))`

> Which rule, if any, is this issue related to?

function-calc-no-unspaced-operator

> What CSS is needed to reproduce this issue?

```css
.foo {
  top: calc(-var(--bar))
}
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
  "rules": {
    "function-calc-no-unspaced-operator": ["always"]
  }
}
```

> Which version of stylelint are you using?

7.1.0

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Using `gulp-stylelint`:

```js
gulp.task('styles-lint', () => (
  gulp.src(cssSrc)
  .pipe(stylelint({
    failAfterError: true,
    reporters: [
      { formatter: 'verbose', console: true },
    ],
  }))
));
```

> What did you expect to happen?

Pass with no problem

> What actually happened (e.g. what warnings or errors you are getting)?

Get the following error:

```shell
 57:19  ✖  Expected single space before "-" operator   function-calc-no-unspaced-operator
 57:19  ✖  Expected single space after "-" operator    function-calc-no-unspaced-operator
```

Not sure what the best way to solve this issue is however...

**Comment by jeddy3** [comment]
@perrin4869 Thanks for writing up the issue using the template.

Is `calc(-var(--foo))` valid syntax? I couldn't find any examples of it in the specs?

As an aside, is the explicitness of `calc(var(--foo) * -1)` not preferred?

**Comment by perrin4869** [comment]
@jeddy3 Really not sure if it's valid sytax, but it compiles with `cssnext` postcss plugin like you'd expect, so I am using it.

That being said, `-var(--foo)` is explicit enough in my opinion. You wouldn't write `3 * (-1)` instead of `-3`. I guess it's more declarative and cleaner in my opinion...

**Comment by perrin4869** [comment]
hm... I could put together a PR soon enough. Maybe the best way to proceed is to check if after a `minus` or a `plus` there is a `var` function, and if so, set `afterOk` and `beforeOk` to `true`?

**Comment by davidtheclark** [comment]
@perrin4869 @jeddy3 I do not believe it is valid syntax, and I think if a PostCSS plugin compiles it that is a mistake on the plugin's part. Let us know if you find evidence in the spec that this is in fact valid.

**Comment by perrin4869** [comment]
ok yeah, looking at the document you are probably right. If `calc(var(--foo)em)` doesn't work as opposed to `calc(var(--foo) * 1em)`, I guess the same applies to `-var(--foo)`. Maybe I should raise an issue with `cssnext`. Thanks!
